### DISCUSSION TITLE
Maybe documentation issue: std::size_t is unsigned, but start_object is supposed to pass -1 ?

### DISCUSSION THREAD

**Discussion by mj-64** [nlohmann/json#4246] [General]
The doc for the SAX interface states:

    // called when an object or array begins or ends, resp. The number of elements is passed (or -1 if not known)
    bool start_object(std::size_t elements);
    bool end_object();
    bool start_array(std::size_t elements);
    bool end_array();

std::size_t is an unsigned type, so it cannot be -1. Should that read as -1 = 0xffffffff aka ```std::numeric_limits<std::size_t>::max()``` ?

**Comment by gregmarr** [comment]
It's just that value cast to that type, or `std::size_t(-1)`.

**Reply by nlohmann** [reply]
Exactly.
